presto-main/src/main/java/com/facebook/presto/server/PrestoSystemRequirements.java

```diff
@@ -14,14 +14,20 @@
 package com.facebook.presto.server;
 
 import com.google.common.base.StandardSystemProperty;
+import com.sun.management.UnixOperatingSystemMXBean;
 import io.airlift.slice.Slice;
 import io.airlift.slice.Slices;
 import org.joda.time.DateTime;
 
+import java.lang.management.ManagementFactory;
+import java.lang.management.OperatingSystemMXBean;
 import java.nio.ByteOrder;
 
 final class PrestoSystemRequirements
 {
+
+    private static final int MIN_FILE_DESCRIPTORS = 10_000;
```

Comment:
We set our ulimit to 8192 and we see our threads top out at around 1500 fwiw. Instead of config-driving this, what about logging a warning at startup instead of failing? That way someone running a dev instance locally will be ok.